en,zh: add FAQ about Ready field of TiDBCluster (#1370)
pingcap/docs-tidb-operator: en/faq.md

To ensure data safety in case one node is down, PD and TiKV use [Raft Consistency Algorithm](https://raft.github.io/) to replicate the stored data as multiple replicas across nodes.

In the bottom layer, TiKV replicates data using the log replication and State Machine model. For write requests, data is written to the Leader node first, and then the Leader node replicates the command to its Follower nodes as a log. When most of the Follower nodes in the cluster receive this log from the Leader node, the log is committed and the State Machine changes accordingly.

## If the Ready field of a TidbCluster is false, does it mean that the corresponding TiDBCluster is unavailable?

> After you execute the `kubectl get tc` command, if the output shows that the **Ready** field of a TiDBCluster is false, it does not mean that the corresponding TiDBCluster is unavailable, because the cluster might be in the following status:

Reviewer: ```suggestion
After you execute the `kubectl get tc` command, if the output shows that the **Ready** field of a TiDBCluster is false, it does not mean that the corresponding TiDBCluster is unavailable, because the cluster might be in any of the following status:
```

* Upgrading
* Scaling

> * Any component pod is unready

Reviewer: ```suggestion
* any Pod of PD, TiDB, TiKV, or TiFlash is not Ready
```

> To check whether a TiDB Cluster is unavailable, you can try connecting to TiDB.

Reviewer: ```suggestion
To check whether a TiDB Cluster is unavailable, you can try connecting to TiDB. If the connection fails, it means that the corresponding TiDBCluster is unavailable.
```

To ensure data safety in case one node is down, PD and TiKV use [Raft Consistency Algorithm](https://raft.github.io/) to replicate the stored data as multiple replicas across nodes.

In the bottom layer, TiKV replicates data using the log replication and State Machine model. For write requests, data is written to the Leader node first, and then the Leader node replicates the command to its Follower nodes as a log. When most of the Follower nodes in the cluster receive this log from the Leader node, the log is committed and the State Machine changes accordingly.

> ## Is the cluster unavailable when the Ready field of TiDBCluster is false?

Reviewer: ```suggestion
## If the Ready field of a TiDBCluster is false, does it mean that the corresponding TiDBCluster is unavailable?
```

> After execute `kubectl get tc` command, maybe you notice that the Ready field of TiDBCluster false.
> 
> It doesn't mean that the TiDB cluster is unavailable, maybe cluster is upgrading or scaling. You can check whether cluster is unavailable by connecting TiDB.

Reviewer: ```suggestion
After you execute the `kubectl get tc` command, if the output shows that the **Ready** field of a TiDBCluster is false, it does not mean that the corresponding TiDBCluster is not available, because the cluster might be in the process of upgrading or scaling up.
	
To check whether a TiDBCluster is unavailable, you can try connecting to TiDB.
```

> After you execute the `kubectl get tc` command, if the output shows that the **Ready** field of a TiDBCluster is false, it does not mean that the corresponding TiDBCluster is not available, because the cluster might be in the process of upgrading or scaling up.

Reviewer: ```suggestion
After you execute the `kubectl get tc` command, if the output shows that the **Ready** field of a TiDBCluster is false, it does not mean that the corresponding TiDBCluster is unavailable, because the cluster might be in the process of upgrading or scaling up.
```
